ReactCommon/react/renderer/core/EventListener.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "EventListener.h"
+
+namespace facebook::react {
+
+bool EventListeners::callListeners(
+    const EventTarget *eventTarget,
+    const std::string &type,
+    ReactEventPriority priority,
+    const ValueFactory &payloadFactory) {
+  std::shared_lock<butter::shared_mutex> lock(lock_);
+
+  bool result = true;
+  for (auto const &listener : eventListeners_) {
+    result =
+        result && listener->operator()(eventTarget, type, priority, payloadFactory);
+  }
+  return result;
```

Comment:
If I understand it correctly, all event listeners intercept the event (in the order of registration) and each one of them individually returns a boolean value whether the event should be passed to JS or not. If any of the event listeners returns `false`, the event will not be passed to JS.

Also, even if some event listener returns `false` (meaning it could handle the event), the rest of event listeners will still be called, which seems fair. (This guarantee will prevent third-party library developers racing for first place as there is no priority in voting).

I know it's a matter of preference, but personally it makes more sense to me to return `true` from event listener if it successfully managed to handle the event (so it doesn't need to be sent to JS) and `false` otherwise (meaning it couldn't handle the event so it probably needs to be passed to JS). So we would start from `result = false`, use `||` instead of `&&` here and don't need `!` in `Scheduler.cpp`.

Also, a comment on the semantics of the return value from event listener would be helpful for the other developers as well.

BTW I'm not sure but it looks like the syntax without `->operator()`, i.e. `listener(eventTarget, type, priority, payloadFactory)` would also work.

Reply:
Agree on true/false comment, it will be more aligned with platform apis that way
I added the comment for the Event listener internally, but it probably didn't sync to GitHub.

Also about `operator()`, it is dereferencing shared pointer here, so just calling the function doesn't work, unfortunately
I could deref it into `EventListener const &` and call it that way, but not sure there is a point in doing so

Reply:
You're right, I forgot that it's a shared pointer. In such case `(*listener)(eventTarget, type, priority, payloadFactory)` would do, but the original version with `->operator()` is certainly better in terms of explicitness and readability.